### DISCUSSION TITLE
Cannot install package from github package registry

### DISCUSSION THREAD

**Discussion by spacefugitive** [community/community#25680] [Actions]
<p>I have an npm package uploaded to a organisation repo. Let’s call my org <code>Org</code>; the repo is <code>kafka-client</code>. (the first letter of my Org is uppercased)</p>
<p>I have two sets of errors:</p>
<ol>
<li>When I change the org scope (<span class="mention">@OWNER</span>) to <code>@Org</code> (first letter upcased) as per the documentation: <a href="https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-npm-registry#authenticating-to-github-packages" rel="noopener nofollow ugc">working with NPM registry</a>; I get the following error:</li>
</ol>
<p>.npmrc</p>
<pre><code class="lang-auto">npm.pkg.github.com/:_authToken=${NPM_TOKEN}
@Orgname:registry=https://npm.pkg.github.org
</code></pre>
<p>error trace:</p>
<pre><code class="lang-auto">npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/@org%2fkafka-client - Not found
npm ERR! 404
npm ERR! 404  '@org/kafka-client@0.0.1' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.
</code></pre>
<p>Not sure know why it looks up <code>registry.npmjs.org</code> registry as opposed to <code>npm.pkg.github.org</code>; assuming it could not find the package there.</p>
<p>Whats weird is when I switch the case of the org to lowercase I get a whole different error:<br>
.npmrc</p>
<pre><code class="lang-auto">npm.pkg.github.com/:_authToken=${NPM_TOKEN}
@orgname:registry=https://npm.pkg.github.org
</code></pre>
<p>error trace:</p>
<pre><code class="lang-auto">npm ERR! code ENOTFOUND
npm ERR! syscall getaddrinfo
npm ERR! errno ENOTFOUND
npm ERR! network request to https://npm.pkg.github.org/@org%2fkafka-client failed, reason: getaddrinfo ENOTFOUND npm.pkg.github.org
npm ERR! network This is a problem related to network connectivity.
npm ERR! network In most cases you are behind a proxy or have bad network settings.
npm ERR! network
npm ERR! network If you are behind a proxy, please make sure that the
npm ERR! network 'proxy' config is set properly.  See: 'npm help config'
</code></pre>
<p>The package is there in the repo; From these logs I don’t know how to proceed. Anyone know what could be wrong?</p>
<p>I have no proxy / http_proxy settings configured for npm</p>

**Comment by toddself** [comment, marked as answer]
<p>The first thing I notice is your .npmrc is set up for <code>@Orgname</code> but the logs you’re showing are <code>@org/kafka-client</code>.  Can you confirm that you’re installing <code>@Orgname/kafka-client</code>.</p>
<p>The npm cli will speak to the default registry <code>registry.npmjs.com</code> when it doesn’t find a matching scope that is set to an alternate registry.</p>
<p>The second issue is because the URL you should be using is <code>npm.pkg.github.com</code> — using <code>.org</code> will definitely fail with the message you’ve got because that domain name doesn’t exist :slight_smile:</p>

**Comment by spacefugitive** [comment]
<p>Thank you for pointing this out.  I’m obviously blind; and not very smart.</p>
